test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -0,0 +1,495 @@
+            ((7, 9, 10, 16), 16, 16),
+            ((7, 9, 10, 16), 10, 10),
+            ((7, 9, 10, 16), 9, 9),
+            ((7, 9, 10, 16), 8, 9),
+            ((7, 9, 10, 16), 7, 7),
+            ([17], 16, None),  # New op added in 17
+            ([9], 9, 9),
+            ([9], 8, 9),
+            ([], 16, None),
+            ([], 9, None),
+            ([], 8, None),
+            # Ops registered at opset 1 found as a fallback when target >= 9
+            ([1], 16, 1),
+        ],
+    )
+    def test_dispatch_opset_version_returns_correct_version(
+        self, available_opsets: Sequence[int], target: int, expected: int
+    ):
+        actual = onnxfunction_dispatcher._dispatch_opset_version(
+            target, available_opsets
+        )
+        self.assertEqual(actual, expected)
+
+
+class TestOpSchemaWrapper(common_utils.TestCase):
+    def setUp(self):
+        # OnnxFunction with default attributes
+        self.onnx_function_add = ops.core.aten_add
+        self.op_schema_wrapper_add = onnxfunction_dispatcher._OpSchemaWrapper(
+            self.onnx_function_add.op_schema
+        )
+
+        # overload type: overloaded dtypes
+        self.onnx_function_mul = ops.core.aten_mul
+        self.onnx_function_mul_bool = ops.core.aten_mul_bool
+
+        # overload type: optional dtype
+        self.onnx_function_new_full = ops.core.aten_new_full
+        self.onnx_function_new_full_dtype = ops.core.aten_new_full_dtype
+
+    def test_perfect_match_inputs(self):
+        inputs = [torch.randn(3, 4), torch.randn(3, 4)]
+        wrong_type_inputs = ["A", "B"]
+        wrong_number_of_inputs = [
+            torch.randn(3, 4),
+            torch.randn(3, 4),
+            torch.randn(3, 4),
+        ]
+        self.assertTrue(
+            self.op_schema_wrapper_add.perfect_match_inputs(inputs, {"alpha": 2.0})
+        )
+        # Even though the attributes has default, to pass perfect match, the attributes must be specified
+        # Otherwise, we rely on the matching score mechanism
+        self.assertFalse(
+            self.op_schema_wrapper_add.perfect_match_inputs(wrong_type_inputs, {})
+        )
+        self.assertFalse(
+            self.op_schema_wrapper_add.perfect_match_inputs(wrong_number_of_inputs, {})
+        )
+        # wrong kwargs
+        self.assertFalse(
+            self.op_schema_wrapper_add.perfect_match_inputs(
+                inputs, {"wrong_kwargs": 2.0}
+            )
+        )
+
+    def test_matching_score_system_on_overload_dtypes(self):
+        inputs = [torch.randn(3, 4), torch.randn(3, 4)]
+        inputs_bool = [
+            torch.randint(0, 2, size=(3, 4), dtype=torch.int).bool(),
+            torch.randint(0, 2, size=(3, 4), dtype=torch.int).bool(),
+        ]
+
+        op_schema_wrapper_mul = onnxfunction_dispatcher._OpSchemaWrapper(
+            self.onnx_function_mul.op_schema
+        )
+        op_schema_wrapper_mul._record_matching_score(inputs, {})
```

Comment:
Why is `_record_matching_score` private and being used here?

Reply:
I wanted to test the private function.